### DISCUSSION TITLE
"Unable to fetch data" error on Visualize tab, all graphs ok on Home tab

### DISCUSSION THREAD

**Discussion by kornix** [akvorado/akvorado#1165] [Q&A]
Hi All.

Have an standalone (no docker) Akvorado setup on a server running Rocky 9.3 / Go 1.22.2 / Clickhouse 24.3.2.23 / Kafka 3.7.0.

After upgrading Akvorado from 1.9.3 to 1.10.0 (applying all required migrations and schemas), i can see all summary graphs on Home tab, but no detailed graph on Visualize tab - just a red error message "Unable to fetch data! Server returned an error: Error" instead (see attached screenshots).

Clickhouse shows error containing current Dimension used to build the graph like that:

```
Row 5:
──────
last_error_time:    2024-04-09 12:35:30
last_error_message: Not found column __table5.ExporterAddress
                                              ^^^^^^^^^^^^^^^^^
```

All flows tables seems to be created after orchestrator restart, but the error looks like they are not:

    ┌─name──────────────────────────────────────────┐
 1. │ asns                                          │
 2. │ exporters                                     │
 3. │ exporters_consumer                            │
 4. │ flows                                         │
 5. │ flows_1h0m0s                                  │
 6. │ flows_1h0m0s_consumer                         │
 7. │ flows_1m0s                                    │
 8. │ flows_1m0s_consumer                           │
 9. │ flows_5m0s                                    │
10. │ flows_5m0s_consumer                           │
11. │ flows_LAABIGYMRYZPTGOYIIFZNYDEQM_raw          │
12. │ flows_LAABIGYMRYZPTGOYIIFZNYDEQM_raw_consumer │
13. │ flows_raw_errors                              │
14. │ flows_raw_errors_consumer                     │
15. │ icmp                                          │
16. │ networks                                      │
17. │ protocols                                     │
    └───────────────────────────────────────────────┘

Any ideas where i should take a look?

Thank you in advance.

![akvorado1](https://github.com/akvorado/akvorado/assets/16319054/114d8fe7-2042-49f7-8343-146ce578261d)
![akvorado2](https://github.com/akvorado/akvorado/assets/16319054/40d9cd17-9cd8-4a4c-ab82-fc1569016364)

**Comment by kornix** [comment]
A little update: Sankey graph type works ok with the same (problematic) dimensions (e. g. when taking SrcAS vs. DstAS etc.), other graph types throws error.

**Reply by vincentbernat** [reply]
On your browser, in the console, it displays the raw SQL request that is executed. Could you execute it by hand? The `__table5` is a bit odd.

**Reply by kornix** [reply]
X-Sql-Query header contains:

WITH source AS (SELECT * FROM flows_5m0s SETTINGS asterisk_include_alias_columns = 1), rows AS (SELECT SrcAS FROM source WHERE TimeReceived BETWEEN toDateTime('2024-04-08 04:50:00', 'UTC') AND toDateTime('2024-04-09 16:50:00', 'UTC') GROUP BY SrcAS ORDER BY SUM(Bytes) DESC LIMIT 10) SELECT 1 AS axis, * FROM ( SELECT toStartOfInterval(TimeReceived + INTERVAL 600 second, INTERVAL 600 second) - INTERVAL 600 second AS time, SUM(Bytes*SamplingRate*8)/600 AS xps, if((SrcAS) IN rows, [concat(toString(SrcAS), ': ', dictGetOrDefault('asns', 'name', SrcAS, '???'))], ['Other']) AS dimensions FROM source WHERE TimeReceived BETWEEN toDateTime('2024-04-08 04:50:00', 'UTC') AND toDateTime('2024-04-09 16:50:00', 'UTC') GROUP BY time, dimensions ORDER BY time WITH FILL FROM toDateTime('2024-04-08 04:50:00', 'UTC') TO toDateTime('2024-04-09 16:50:00', 'UTC') + INTERVAL 1 second STEP 600 INTERPOLATE (dimensions AS ['Other']))

Clickhouse returns:

`Query id: 0cd9e130-03d4-48b8-a457-1f030ef02c48

Elapsed: 0.009 sec. 

Received exception from server (version 24.3.2):
Code: 10. DB::Exception: Received from localhost:XXX. DB::Exception: Not found column __table5.SrcAS: in block minus(toStartOfInterval(plus(__table5.TimeReceived, _CAST(600_IntervalSecond, 'IntervalSecond'_String)), _CAST(600_IntervalSecond, 'IntervalSecond'_String)), _CAST(600_IntervalSecond, 'IntervalSecond'_String)) DateTime UInt32(size = 0), divide(multiply(sum(multiply(__table5.Bytes, __table5.SamplingRate)), 8_UInt8), 600_UInt16) Float64 Float64(size = 0), if(in(__table5.SrcAS, __set_16562756643659182990_17750305686501399710), array(concat(toString(__table5.SrcAS), ': '_String, dictGetOrDefault('asns'_String, 'name'_String, __table5.SrcAS, '???'_String))), _CAST(['Other']_Array(String), 'Array(String)'_String)) Array(String) Array(size = 0, UInt64(size = 0), String(size = 0)). (NOT_FOUND_COLUMN_IN_BLOCK)
`

Akvorado returns 500 Internal Server Error code and {"message":"Unable to query database."} response.

Should i try to drop Clickhouse DB and restart orchestrator to recreate it from the scratch?

**Reply by vincentbernat** [reply]
What is your version of ClickHouse?

**Reply by vincentbernat** [reply]
Nevermind, you provided it. Your version is a bit recent. I suppose you were not running that with 1.9.3? I need to check with this version as well.

**Reply by kornix** [reply]
No. Akvorado 1.9.3 was used with Clickhouse 22.6. I've updated environment required by Akvorado (golang, clickhouse, kafka and zookeeper) to the last stable/recommended versions before updating Akvorado from 1.9.3 to 1.10.0. All system updates are applied on a regular basis by dnf-automatic, so the system runs up-to-date software since it's installation.

**Reply by vincentbernat** [reply]
I get the same problem with ClickHouse 24.3. Keep 23.8 for now.

**Reply by kornix** [reply]
Downgraded Clickhouse to 23.12. This made the trick.

Very appreciate for your help.

**Reply by vincentbernat** [reply]
See https://github.com/ClickHouse/ClickHouse/issues/62464.

**Reply by FestplattenSchnitzel** [reply]
Works for me after upgrading Clickhouse to 24.5.1.1763, which seems to contain the linked fix.

**Comment by kornix** [comment]
Problem with Akvorado v1.10.0 Visualize tab graphs was solved when downgraded Clickhouse to 23.12.
